### PR TITLE
Provide `Microsoft.Extensions.Http.Polly` package README

### PR DESCRIPTION
**Pull request by MackinnonBuck**
Contributes to #48392

### CODE REVIEW

File: src/HttpClientFactory/Polly/src/PACKAGE.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        httpRequestMessage.Method == HttpMethod.Get
+            ? timeoutPolicy
+            : longTimeoutPolicy);
+```
+
+In this example, if the outgoing request is an HTTP GET, a 10-second timeout is applied. For any other HTTP method, a 30-second timeout is used.
+
+#### Additional scenarios
+
+For more examples on using Polly-based handlers, see the [official documentation](https://learn.microsoft.com/aspnet/core/fundamentals/http-requests?view#use-polly-based-handlers).
```
Comment (halter73) [inline src/HttpClientFactory/Polly/src/PACKAGE.md:52]: Nit: This seems a little too similar to the "Additional Documentation".

```suggestion
```

**Review by halter73 — APPROVED** [review]
LGTM. @martincostello Do have any thoughts?

**Comment by martincostello** [comment]
> Do have any thoughts?

The document looks fine to me as-is.

One thought I had thought I had though was _maybe_ it would be worth a comment encouraging people to use to the new resilience libraries from dotnet/extensions that use the new Polly.Core APIs? There have been a few issues where people have asked when this package will be updated to use Polly v8 and I've had to explain the sort-of fork that's happened and that this package is kinda frozen and the new innovation is in the new extensions package.

**Review by martincostello — APPROVED** [review]

**Comment by MackinnonBuck** [comment]
> One thought I had thought I had though was maybe it would be worth a comment encouraging people to use to the new resilience libraries from dotnet/extensions that use the new Polly.Core APIs?

That's a good idea. I've updated the README to include such a note. Some of our [docs](https://learn.microsoft.com/dotnet/core/resilience) even go as far as to say that `Microsoft.Extensions.Http.Polly` is deprecated, although the [ASP.NET Core docs](https://learn.microsoft.com/aspnet/core/fundamentals/http-requests#use-polly-based-handlers) don't mention this, and we're still shipping new versions. Maybe we should consider not shipping new versions of this package (and updating the ASP.NET Core docs).

**Comment by martincostello** [comment]
> Maybe we should consider not shipping new versions of this package (and updating the ASP.NET Core docs).

Maybe for .NET 10? It's a non-trivial piece of work to swap from one to the other, so at this stage it would be a bit of annoying rug-pull for users.

**Pull request merged**